### ISSUE TITLE
Working with nullable properties.

### ISSUE THREAD

**Issue by todo-username** [laminas/laminas-code#196]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | [4.13.0](https://github.com/laminas/laminas-code/releases/tag/4.13.0)

#### Summary

When creating methods with arguments that accept NULL, an additional '\\' character is inserted.

![image](https://github.com/laminas/laminas-code/assets/43653056/f5602c26-5fa9-4c31-9104-20259a1fcd5d)

#### Current behavior

When creating methods with arguments that accept NULL, an additional '\\' character is inserted.

![image](https://github.com/laminas/laminas-code/assets/43653056/f5602c26-5fa9-4c31-9104-20259a1fcd5d)

This behavior is most likely due to this method

src/Generator/TypeGenerator/AtomicType.php
![image](https://github.com/laminas/laminas-code/assets/43653056/9934e949-c3cf-499e-817c-a53d0e810a2d)

If you remove the substitution of these characters, the method is generated correctly
![image](https://github.com/laminas/laminas-code/assets/43653056/5cf9dab3-481c-4906-8f70-a656c15d5f27)

Most likely, these characters are removed incorrectly if use ClassName is present in the class. If you create properties without the '?' The extra '\\' is not added.

#### How to reproduce

Try creating methods for nullable properties (not primitives).
![image](https://github.com/laminas/laminas-code/assets/43653056/d85088ee-bbac-4b11-95ee-4075ecf7327a)

#### Expected behavior

The method is created without the additional '\\' character.

![image](https://github.com/laminas/laminas-code/assets/43653056/53e6ebdb-5d60-4f41-b53c-59a2b38fb13e)

**Comment by Ocramius** [comment]
Are you able to write a snippet that runs the generator and reproduces the problem here? I can't understand how `laminas/laminas-code` is used in your example.

Also, `\ArrayCollection` is correct if `'ArrayCollection'` is given as string. All class names are assumed to be FQCNs, and I'd say that the property not having the `\` is more problematic there.

**Comment by todo-username** [comment]
> Also, `\ArrayCollection` is correct if `'ArrayCollection'` is given as string.

@Ocramius Yes, it is passed as a string, because if you pass the full class path, an additional prefix will be written, which is not needed due to the use of 'use'

![image](https://github.com/laminas/laminas-code/assets/43653056/0ce1893d-5a72-498d-9389-526e8ddedc06)

**Comment by todo-username** [comment]
> Are you able to write a snippet that runs the generator and reproduces the problem here? I can't understand how `laminas/laminas-code` is used in your example.

@Ocramius Can you give an example of a format that will make it easier for you to understand the problem?

**Comment by Ocramius** [comment]
> > Also, `\ArrayCollection` is correct if `'ArrayCollection'` is given as string.
> 
> @Ocramius Yes, it is passed as a string, because if you pass the full class path, an additional prefix will be written, which is not needed due to the use of 'use'

`laminas/laminas-code` explicitly does **NOT** consider `use` statements: all symbols are FQN (fully qualified name) references.

More in https://github.com/laminas/laminas-code/issues/176

 > Can you give an example of a format that will make it easier for you to understand the problem?

The best you can do to reproduce the problem is:

 1. clone this repository
 2. make an example for `ClassGenerator` (see https://github.com/laminas/laminas-code/blob/fd87696ee3c03073058ee26176d0a7049930fefe/test/Generator/ClassGeneratorTest.php)
 3. make it fail with your scenario, to highlight the expectation differences

BTW, don't paste in screenshots: they are terrible to process :D You can use triple backticks for a code snippet.

That said, I endorse checking #176 first.
